artifactory/commands/transferfiles/utils.go

```diff
@@ -149,18 +153,18 @@ var processedUploadChunksMutex sync.Mutex
 // Number of chunks is limited by the number of threads.
 // Whenever the status of a chunk was received and is DONE, its token is removed from the tokens batch, making room for a new chunk to be uploaded
 // and a new token to be polled on.
-func pollUploads(phaseBase *phaseBase, srcUpService *srcUserPluginService, uploadChunkChan chan UploadedChunkData, doneChan chan bool, errorsChannelMng *ErrorsChannelMng) {
-	curTokensBatch := UploadChunksStatusBody{}
+func pollUploads(phaseBase *phaseBase, srcUpService *srcUserPluginService, uploadChunkChan chan UploadedChunk, doneChan chan bool, errorsChannelMng *ErrorsChannelMng) {
```

Comment:
I syggest moving this function, and possibly the functions it uses, into the `transferfiles/manager.go` file. This is because it looks like this function is used only by that file (and the `transfer_test.go` file).
It is best to reduce the code in the util files, and keep there only code which is used a util (shared between multiple components).